### DISCUSSION TITLE
旧版可以爬取任意用户的微博内容，改为使用cookies后好像不行了？

### DISCUSSION THREAD

**Discussion by HayashiKaori** [Chilfish/Weibo-archiver#24] [General]
我使用自己账号的cookies，下载其他账号的图片都失败了。
旧版输入爬取用户的UID是可以的，能支持么？
因为备份任意用户（比如字幕组）的微博，也是很常用的需求。

**Comment by Chilfish** [comment]
我试着是可以的欸，也是用 `document.cookie`，然后复制内容（不包括双引号）、然后下载别人的图片

**Reply by Chilfish** [reply]
包含 `XSRF-TOKEN=xx; ALF=xxx; SUBP=xx.xx.xx` 这样格式的内容基本就可以的

**Reply by Chilfish** [reply]
emm如果还是不行，可以手动修改 download.mjs 里的 download 函数的

https://github.com/Chilfish/Weibo-archiver/blob/4328e2c2beb15e3c03e15acca7bb31ff8fa9fb2a/scripts/download.mjs#L77

成下面这样

```js
const headers = {
  Cookie: cookie,
  referrer: 'https://weibo.com/',
}
```

后来我才发现 uid 不是必须的，只要有这个 referrer 指向微博的域名就行了😹 @HayashiKaori

**Reply by Chilfish** [reply]
于是现在的逻辑改为了：只有从旧版导出的 imgs.csv 才需要 cookie 下载评论图片，新版什么都不需要，直接下载就行了（因为已经导出了正确的图片评论的链接，不需要 cookie 获取了）
